src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -33,81 +45,314 @@
+     * @return the ResponseEntity with status 200 (OK) and with body the fileUploadSubmission, or with status 404 (Not Found)
      */
-    @PutMapping("/file-upload-submissions")
-    public ResponseEntity<FileUploadSubmission> updateFileUploadSubmission(@RequestBody FileUploadSubmission fileUploadSubmission) throws URISyntaxException {
-        log.debug("REST request to update FileUploadSubmission : {}", fileUploadSubmission);
-        if (fileUploadSubmission.getId() == null) {
-            return createFileUploadSubmission(fileUploadSubmission);
-        }
-        FileUploadSubmission result = fileUploadSubmissionRepository.save(fileUploadSubmission);
-        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, fileUploadSubmission.getId().toString())).body(result);
+    @GetMapping("/file-upload-submissions/{submissionId}")
+    public ResponseEntity<FileUploadSubmission> getFileUploadSubmission(@PathVariable Long submissionId) {
+        log.debug("REST request to get FileUploadSubmission with id: {}", submissionId);
+        var fileUploadExercise = (FileUploadExercise) fileUploadSubmissionService.findOne(submissionId).getParticipation().getExercise();
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(fileUploadExercise)) {
+            return forbidden();
+        }
+        var fileUploadSubmission = fileUploadSubmissionService.getLockedFileUploadSubmission(submissionId, fileUploadExercise);
+        // Make sure the exercise is connected to the participation in the json response
+        StudentParticipation studentParticipation = (StudentParticipation) fileUploadSubmission.getParticipation();
+        studentParticipation.setExercise(fileUploadExercise);
+        hideDetails(fileUploadSubmission);
+        return ResponseEntity.ok(fileUploadSubmission);
     }
 
     /**
-     * GET /file-upload-submissions : get all the fileUploadSubmissions.
+     * GET /file-upload-submissions : get all the fileUploadSubmissions for an exercise. It is possible to filter, to receive only the one that have been already submitted, or only the one
+     * assessed by the tutor who is doing the call
      *
-     * @return the ResponseEntity with status 200 (OK) and the list of fileUploadSubmissions in body
+     * @param exerciseId the id of the exercise
+     * @param submittedOnly if only submitted submissions should be returned
+     * @param assessedByTutor if the submission was assessed by calling tutor
+     * @return the ResponseEntity with status 200 (OK) and the list of File Upload Submissions in body
      */
-    @GetMapping("/file-upload-submissions")
-    public List<FileUploadSubmission> getAllFileUploadSubmissions() {
-        log.debug("REST request to get all FileUploadSubmissions");
-        return fileUploadSubmissionRepository.findAll();
+    @GetMapping("/exercises/{exerciseId}/file-upload-submissions")
```

Comment:
for which use case do we need this API endpoint?

Reply:
This is used to list submission on the exercise dashboard to the assessor (started and completed)